docs/core/allocation.qmd

```diff
@@ -0,0 +1,178 @@
+$$  {#eq-flowconservationconstraint}
+Note that we do not require equality here; in the allocation we do not mind that excess flow is 'forgotten' if it cannot contribute to the allocation to the users.
+$\Phi_{\hat{k}}$ is the local water supply of the basins:
+$$
+    \Phi_{\hat{k}} = \max\left(\phi_k(t) + \frac{u_k(t)-S(l_{\min,k})}{\Delta t_\text{alloc}}, 0.0 \right).
+$$ {#eq-basinsourceflow}
+Here the first term denotes the vertical fluxes and the second term the flow that can be supplied by the water in the basin above its minimum level, where $\Delta t_\text{alloc}$ is the allocation solve timestep.
+
+- Capacity: the flows over the edges are positive and bounded by the edge capacity:
+$$
+    F_{\hat{i}\hat{j}} \le \left(\hat{C}_S\right)_{\hat{i}\hat{j}}, \quad \forall(\hat{i},\hat{j}) \in \hat{E}_S.
+$$ {#eq-capacityconstraint}
+- User outflow: The outflow of the user is dictated by the inflow and the return factor:
+$$
+\begin{align}
+    F_{\hat{i}\hat{k}} = r_k \cdot F_{\hat{k}\hat{j}}\\
+    \quad \forall\hat{k} \in \hat{U}_s, \\
+    \hat{V}^{\text{in}}_S(\hat{k}) = \{\hat{i}\},\\
+     \hat{V}^{\text{out}}_S(\hat{k}) = \{\hat{j}\}.
+\end{align}
+$$ {#eq-returnflowconstraint}
+Here we use that each user node in the allocation graph has an unique in-edge and out-edge.
+- User allocation: The flow over the edge to the user is equal to the sum of the allocations to the user:
+$$
+    F_{\hat{i}\hat{k}} = \sum_{p=1}^{p_\max} A^\text{user}_{\hat{k},p}, \quad \forall \hat{k} \in \hat{U}_S, \hat{V}^{\text{out}}_s(\hat{k}) = \{\hat{i}\}.
+$$ {#eq-userallocationconstraint}
+Here we use that each user has an unique out-edge.
+- User demand: what is allocated to the user is bounded above by the user demand:
+$$
+    A_{\hat{i},p}^\text{user} \leq d_i^p(t) \quad \forall\hat{i} \in \hat{U}_S, \; p = 1,\ldots,p_\max.
+$$
+- Basin allocation: If the flow supplied by a basin (as determined in @eq-basinsourceflow) is negative, it is a demand:
+$$
+A^\text{basin}_{\hat{i}} = \max\left(-\left(\phi_i(t) + \frac{u_i(t)-S(l_{\min,k})}{\Delta t_\text{alloc}}\right), 0.0 \right), \quad \forall \hat{i} \in \hat{B}_S.
+$$ {#eq-basinallocationconstraint}
+- Fractinal flow: Let $\hat{L}_S \subset \hat{V}_S$ be the set of nodes in the max flow graph with fractional flow outneighbors, and $f_j$ the flow fraction associated with fractional flow node $j \in V_S$. Then
+$$
+\begin{align}
+    F_{\hat{i}\hat{j}} = f_j \sum_{k\in \hat{V}^\text{in}_S(\hat{i})} F_{\hat{k}\hat{i}} \\
+    \forall \hat{i} \in \hat{L}_S, \\
+    \hat{j} \in \hat{V}_S^\text{out}(\hat{i}).
+\end{align}
+$$ {#eq-fractionalflowconstraint}
+
+- Flow sign: Furthermore there are the non-negativity constraints for the flows and allocations, see [The optimization variables](allocation.qmd#the-optimization-variables).
+
+:::{.callout-note}
+Note that $\Phi_{\hat{k}}$ can become negative, in which case @eq-flowconservationconstraint becomes impossible to satisfy. What to do in this case?
```

Comment:
How is this possible, isn't it defined to be `max(x, 0.0)`?

Reply:
You're right, I forgot to remove this after I found a solution for it.